third_party/openthread/platforms/BUILD.gn

```diff
@@ -45,19 +47,24 @@ static_library("libopenthread-platform-utils") {
     "${openthread_root}/examples/platforms/utils/code_utils.h",
     "${openthread_root}/examples/platforms/utils/debug_uart.c",
     "${openthread_root}/examples/platforms/utils/encoding.h",
-    "${openthread_root}/examples/platforms/utils/logging_rtt.c",
-    "${openthread_root}/examples/platforms/utils/logging_rtt.h",
     "${openthread_root}/examples/platforms/utils/mac_frame.cpp",
     "${openthread_root}/examples/platforms/utils/mac_frame.h",
     "${openthread_root}/examples/platforms/utils/settings.h",
     "${openthread_root}/examples/platforms/utils/settings_ram.c",
     "${openthread_root}/examples/platforms/utils/soft_source_match_table.c",
     "${openthread_root}/examples/platforms/utils/soft_source_match_table.h",
   ]
-  public_deps = [
-    "${openthread_root}/src/core:libopenthread_core_headers",
-    "${segger_rtt_root}:segger_rtt",
-  ]
+
+  public_deps = [ "${openthread_root}/src/core:libopenthread_core_headers" ]
+
+  if (chip_enable_segger_rtt == true) {
```

Comment:
is the `== true` needed? Generally you want `if (foo)` or `if (!foo)` instead of `if (foo == true)` or `if (foo == false)`

Reply:
Yes, we can do that. It’s easy to miss ! tough, especially if you are tired or in a hurry.
I was trying to follow the style already present, and in:
src/platform/device.gni
on line 183, we have:
~~~
assert(chip_disable_platform_kvs == false || chip_device_platform == "darwin",
~~~

Reply:
the == false is odd there. General code review guideline would be to to not use if with == false/==true.

I don't think anyone will mis-read `if (chip_enable_segger_rtt)`. Lots of examples for if conditions:

https://github.com/project-chip/connectedhomeip/blob/master/src/platform/Linux/BUILD.gn#L126

And trying to see how much the pattern is used:

```
> rg --glob '*.gn*' 'if \(chip_[a-z_]*\)' --files-with-matches | wc -l
42
```

Reply:
We also have two counter examples. Those are wrong and should be fixed. We need very boring consistency in our code :) 

```
rg --glob '*.gn*' 'if \(chip_[a-z_]* == (true|false)\)'
system/system.gni
60:  } else if (chip_system_config_use_dispatch == false) {

platform/Darwin/BUILD.gn
84:  if (chip_disable_platform_kvs == false) 
```

it does look like maybe this was done for more clarity in the not case. But `rg --glob '*.gn*' 'if \(!chip_[a-z_]*\)'` gives me 4 matches using `!` ...

Reply:
created #23814 to undo the == false from the other places

Reply:
Thank you, I will address review remarks.